API improvement for remove, permanent remove and restore features

olehkorkh-planeks opened the pull request:
- Added new API endpoints
- Added new model fields
- Improved `settings.py`
- Added Celery support

For running Celery you can use the next command:

```shell
celery -A mcserver worker -l INFO -B
```

antoinefalisse reviewed (approved):
@AlbertoCasasOrtiz do you want to take a look? Same here, hard for me to review. I would be tempted to merge and test, but maybe you want to take a look before we do. Do we need to apply the migration after merging in dev?

antoinefalisse commented on mcserver/settings.py:
```diff
@@ -205,3 +206,58 @@
+        send_default_pii=True
+    )
+
+REDIS_URL = config('REDIS_URL')
```
@olehkorkh-planeks looks like I should add this to my .env variable, what should it be?

antoinefalisse commented:
@olehkorkh-planeks thanks!

**Two questions:**
1. See my comment above, but it looks like I should have a 'REDIS_URL' variable in my .env, which I don't have. What should I use?
2. Naive question but I am not sure what you mean with 'For running Celery you can use the next command: `celery -A mcserver worker -l INFO -B`'. Do we need to "manually" run celery? Is that for testing only? Thanks for elaborating on that.

olehkorkh-planeks commented:
Hi @antoinefalisse 

You're right. You need to add the `REDIS_URL` to the `.env` file.

As you can see, I added the Celery as a dependency to the backend. For the current moment, we use it for running background tasks for permanent cleaning of trashed objects (session and trials). We also use Celery Beat for running those tasks by schedule. So, what we need:

- The Redis instance for managing the tasks queue (it is the simplest way for starting with Celery)
- I also added the "celery-redbeat" for managing scheduling metadata in the Redis as well. By default celery beat use a file on the disk for such kind of task, but it is not suitable for cloud hosting.

I wrote about the command for running the celery worker, but it depends on the environment. We need to run a worker and the beat service. You can use the command `celery -A mcserver worker -l INFO -B` for running celery and beat locally. In the production environment, it depends on your infrastructure. You need to check what is suitable in your case.

This is an article about running Celery worker on ECS:

https://python.plainenglish.io/deploying-django-on-aws-setting-up-celery-and-sqs-95632a6e79cb

In my case, I prefer to use Redis instead of SQS, because we can use Redis also for scheduling metadata, and it is possible to use it for caching in the future. SQS can be better for systems where background tasks contain some sort of critical operations, but our case is simpler.

Keep in mind that workers can be scalable. It is possible to run any workers in parallel we need (if you need it). But, in that case, you need to avoid use the `-B` (beat) parameter. In that case you need to run additional service specifically for beat: `celery -A mcserver beat`.

suhlrich commented on requirements.txt:
```diff
@@ -1,20 +1,61 @@
+kiwisolver==1.4.4
+kombu==5.2.4
+matplotlib==3.3.4
+numpy==1.21.6
```
why do we need to add all of these packages? Not sure we use numpy, for example, in the API anywhere

olehkorkh-planeks replied:
Good question. I've just added celery and celery-redbeat. I didn't try to review and optimize other requirements. For example, there is a sense of upgrading the versions as well. Because the code just cannot even run on Python 3.11 and requires pretty old 3.7. But I think that it could be an additional task.

suhlrich replied:
Thanks. I see celery and celery-redbeat, but another 40 packages have been added to requirements.txt. Is there a reason for adding them all to the requirements, or can we just add celery and celery-redbeat?

olehkorkh-planeks replied:
That is because I regenerated `requirements.txt` with `pip freeze` command. Your old file did not contain the whole list of dependencies. Many libraries have also their own list of requirements that previously weren't in the file.

antoinefalisse commented:
Thanks for the explanation @olehkorkh-planeks, we will try to deploy to AWS.
